python/src/aac/lang/hierarchy.py

```diff
@@ -0,0 +1,54 @@
+"""Provides utilities to manage and navigate the hierarchy of definitions."""
+
+import logging
+from typing import Optional
+
+from aac.lang.context import ActiveContext
+from aac.lang.definition_helpers import get_definition_by_name, search_definition
+from aac.parser import ParsedDefinition
+
+
+def get_definition_ancestry(definition: ParsedDefinition, context: ActiveContext) -> list[ParsedDefinition]:
+    """Return an ordered list of ancestor definitions starting with the root definition and ending with the argument."""
+
+    ancestors_list = []
+    found_self_defined_ancestor = False
```

Comment:
Yes, currently only `data` is self-defining, but I'm trying to leave some flexibility in here in case someone elected to self-define another structure. That being said, we make some validation assumptions on the fact that `data:` is the base definition, so I haven't thought of a good way to be 100% assumption free.

Reply:
I don't think we'll ever get 100% assumption-free. Although it would be nice, I think it's impractical - even most of what we know about math is based on a few [axioms](https://encyclopediaofmath.org/wiki/Axiomatic_method). I know the end goal is for the AaC language to be self-defining, etc but I think we can only ever approximate that.

Reply:
That's fair. I'll document this assumption as part of my validator 2.0 docmentation.